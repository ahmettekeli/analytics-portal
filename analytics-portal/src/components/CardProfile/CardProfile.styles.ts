import styled from "styled-components";
import { colors } from "../../configs";

type PropType = { isActive: boolean };

export const Wrapper = styled.div<PropType>`

    display: flex;
    justify-content: left;
    align-items: center;
    margin 0.5rem;
    opacity: ${(props) => (props.isActive ? 1 : 0.5)};
`;

export const ProfileImg = styled.img`
  height: 60px;
  padding: 0 0.5rem;
  border-radius: 1rem;
`;

export const NameContainer = styled.div`
  padding: 0.5rem;
  font-weight: bold;
  font-size: 1.2rem;
  color: ${colors.cardText};
  @media (max-width: 630px) {
    margin: 0px;
    line-height: 1;
  }
`;
