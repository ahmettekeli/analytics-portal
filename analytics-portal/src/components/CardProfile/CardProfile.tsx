import capitalize from "lodash/capitalize";
import * as S from "./CardProfile.styles";

function CardProfile({
  isActive,
  src,
  name,
}: {
  isActive: boolean;
  src: string;
  name: string;
}) {
  return (
    <S.Wrapper isActive={isActive}>
      <S.ProfileImg src={src} alt={name} />
      <S.NameContainer>{capitalize(name)}</S.NameContainer>
    </S.Wrapper>
  );
}

export default CardProfile;
