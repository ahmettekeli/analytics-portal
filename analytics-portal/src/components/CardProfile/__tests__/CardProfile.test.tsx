import { cleanup, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import CardProfile from "../CardProfile";

describe("CardProfile", () => {
  let renderResult: ReturnType<typeof render>;
  const src = "http://placeimg.com/640/480/abstract";
  const isActive = true;
  const name = "Test app";

  beforeEach(() => {
    renderResult = render(
      <CardProfile isActive={isActive} src={src} name={name} />
    );
  });
  afterEach(cleanup);

  test("Renders CardProfile", () => {
    expect(renderResult.getByTestId("card-profile")).toBeInTheDocument();
  });

  test("Displays correct image", () => {
    expect(renderResult.getByAltText(name)).toHaveAttribute("src", src);
  });

  test("Displays correct name", () => {
    expect(renderResult.getByText(name)).toBeInTheDocument();
  });

  test("CardProfile matches snapshot", () => {
    const tree = renderer
      .create(<CardProfile isActive={isActive} src={src} name={name} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
