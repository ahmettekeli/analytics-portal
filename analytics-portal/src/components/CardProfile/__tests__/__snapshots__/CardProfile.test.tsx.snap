// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardProfile CardProfile matches snapshot 1`] = `
<div
  className="sc-bdvvtL kFYleh"
  data-testid="card-profile"
>
  <img
    alt="Test app"
    className="sc-gsDKAQ bioFnW"
    src="http://placeimg.com/640/480/abstract"
  />
  <div
    className="sc-dkPtRN cTqKzg"
  >
    Test app
  </div>
</div>
`;
